// angular-paginator/esm2015/lib/angular-paginator.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AngularPaginatorComponent } from './angular-paginator.component';
import { AngularPaginatorPipe } from './pipes/angular-paginator.pipe';
import { AngularPaginatorDirective } from './directives/angular-paginator.directive';
import { AngularPaginatorService } from './services/angular-paginator.service';
import * as ɵngcc0 from '@angular/core';
export class AngularPaginatorModule {
}
AngularPaginatorModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AngularPaginatorModule });
AngularPaginatorModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AngularPaginatorModule_Factory(t) { return new (t || AngularPaginatorModule)(); }, providers: [AngularPaginatorService], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AngularPaginatorModule, { declarations: function () { return [AngularPaginatorComponent, AngularPaginatorPipe, AngularPaginatorDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [AngularPaginatorComponent, AngularPaginatorPipe]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AngularPaginatorModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                providers: [AngularPaginatorService],
                declarations: [AngularPaginatorComponent, AngularPaginatorPipe, AngularPaginatorDirective],
                exports: [AngularPaginatorComponent, AngularPaginatorPipe]
            }]
    }], null, null); })();
