// angular-paginator/esm2015/lib/pipes/angular-paginator.pipe.js
import { Pipe } from '@angular/core';
import { AngularPaginatorService } from '../services/angular-paginator.service';
/**
 * Pagination Pipe which paginates the array for the provided arguments such as `itemsPerPage` and `currentPage`
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/angular-paginator.service';
export class AngularPaginatorPipe {
    /**
     *
     * @param angularPaginatorService service for angular pagiantor
     */
    constructor(angularPaginatorService) {
        this.angularPaginatorService = angularPaginatorService;
    }
    /**
     * this pipes paginates the array for the provided arguments such as `itemsPerPage` and `currentPage`
     *
     * @param array input array for which the manipulation happens
     * @param args input arguments for the paginator pipe
     */
    transform(array, args) {
        const instance = this.createInstance(array, args);
        // create pagination information
        this.angularPaginatorService.registerInstance(instance);
        // set the slicing range
        const start = (instance.currentPage - 1) * instance.itemsPerPage;
        const end = instance.currentPage * instance.itemsPerPage;
        return array.slice(start, end);
    }
    /**
     * create an instance for each pipe
     *
     * @param array input array for which the manipulation happens
     * @param args input arguments for the paginator pipe
     */
    createInstance(array, args) {
        return {
            id: args.id ? args.id : this.angularPaginatorService.id,
            currentPage: args.currentPage ? args.currentPage : 1,
            itemsPerPage: args.itemsPerPage ? args.itemsPerPage : 10,
            totalItems: array.length
        };
    }
}
AngularPaginatorPipe.ɵfac = function AngularPaginatorPipe_Factory(t) { return new (t || AngularPaginatorPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.AngularPaginatorService)); };
AngularPaginatorPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "angularPaginator", type: AngularPaginatorPipe, pure: true });
AngularPaginatorPipe.ctorParameters = () => [
    { type: AngularPaginatorService }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AngularPaginatorPipe, [{
        type: Pipe,
        args: [{
                name: 'angularPaginator'
            }]
    }], function () { return [{ type: ɵngcc1.AngularPaginatorService }]; }, null); })();
