// angular-paginator/esm2015/lib/services/angular-paginator.service.js
import { Injectable, EventEmitter } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class AngularPaginatorService {
    constructor() {
        /** id is used to identify pagination instances */
        this.id = 'ANGULAR_PAGINATOR_DEFAULT';
        /** instances is the collection of pagination componetns */
        this.instances = {};
        /** emits an event with instance id whenever the page number changes */
        this.change = new EventEmitter();
    }
    /**
     * creates an instance for every pipe differentiated with id
     *
     * @param instance each instance refers to a unique pagination component
     */
    registerInstance(instance) {
        if (!this.instances[instance.id]) {
            this.instances[instance.id] = instance;
            this.change.emit(instance.id);
        }
        else {
            const changed = this.updateInstance(instance);
            if (changed) {
                this.change.emit(instance.id);
            }
        }
        return;
    }
    /**
     * updates existing pagination instances available
     * and returns true if value has changed
     *
     * @param instance an unique pagination component
     */
    updateInstance(instance) {
        for (const key in instance) {
            if (instance[key] !== this.instances[instance.id][key]) {
                this.instances[instance.id][key] = instance[key];
                return true;
            }
        }
        return false;
    }
    /**
     * return instance with given id
     *
     * @param id unique id of the pagination component
     */
    getInstance(id) {
        return this.instances[id];
    }
    /**
     * return currentPage for given id
     *
     * @param id unique id of the pagination component
     */
    getCurrentPage(id) {
        return this.instances[id].currentPage;
    }
}
AngularPaginatorService.ɵfac = function AngularPaginatorService_Factory(t) { return new (t || AngularPaginatorService)(); };
AngularPaginatorService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: AngularPaginatorService, factory: AngularPaginatorService.ɵfac });
AngularPaginatorService.ctorParameters = () => [];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AngularPaginatorService, [{
        type: Injectable
    }], function () { return []; }, null); })();
